api/helpers/gamestate/moves/resolve/six.js

```diff
@@ -0,0 +1,50 @@
+module.exports = {
+  friendlyName: 'Resolve Six One-Off',
+
+  description: 'Returns new GameState resulting from resolving an six, which Scrap all Royals and Glasses Eights on the field.',
+
+  inputs: {
+    currentState: {
+      type: 'ref',
+      description: 'The latest GameState before the six scraps all royals and glasses eights',
+      required: true,
+    },
+  },
+  sync: true, // synchronous helper
+  fn: ({ currentState }, exits) => {
+    let result = _.cloneDeep(currentState);
+
+    const player = result.playedBy ? result.p1 : result.p0;
+    const opponent = result.playedBy ? result.p0 : result.p1;
+    
+    let cardToScrap = [];
```

Comment:
Thoughts on the use of `cardsToScrap` as opposed to pushing directly to `result.scap`?

Reply:
not any reason for using it just show it in old resolve six code so i used that. should i make it result.scrap ?

Reply:
Yes I think that's simpler. The main reason it was previously used in the legacy implementation is that each card needed an individual database write and there is a single sails method for doing that in bulk. So we aggregated all of them before writing to the db. 

But in the GameState API we just need to put the right cards into the right lists in memory so there's no need to create the separate array first